src/web/app/pages-instructor/student-list/student-list.component.ts

```diff
@@ -71,19 +79,46 @@ export class StudentListComponent implements OnInit, DoCheck {
               private tableComparatorService: TableComparatorService,
               private ngbModal: NgbModal,
               private differs: IterableDiffers) {
-    this._differ = this.differs.find(this.sections).create();
+    this.sectionDataDiffer = this.differs.find(this.sections).create();
```

Comment:
Personally, I don't like to touch the low-level change detection in Angular as this is prone to errors. Even this is done before, it is still necessary to have one `StudentListSectionData` outside and one `FlatStudentListData` inside?

In other word, can we just construct `FlatStudentListData` (actually we should call it as `StudentListRowModel`) outside?